File: packages/dart/lib/src/nickname/rand_nickname_details.dart
import 'package:randino/src/nickname/nickname_generator.dart';
import 'package:randino/src/types.dart';

/// Generate nicknames along with the pieces each one was built from.
///
/// Takes the same parameters as `randNickname`. Useful when you need the words
/// on their own — to highlight the base word, or to group by theme.
///
/// ```dart
/// randNicknameDetails(language: WordLanguage.ko);
/// // [NicknameDetail(멋진사자, [멋진, 사자], ko, animal)]
/// ```
List<NicknameDetail> randNicknameDetails({
  WordLanguage? language,
  WordTheme? theme,
  int count = 1,
  RandRealism realism = RandRealism.real,
  int? minLength,
  int? maxLength,
  String? wordSeparator,
  String? startsWith,
  bool unique = false,
}) => generateNicknameDetails(
  language: language,
  theme: theme,
  count: count,
  realism: realism,
  minLength: minLength,
  maxLength: maxLength,
  wordSeparator: wordSeparator,
  startsWith: startsWith,
  unique: unique,
);

File: packages/javascript/lib/nickname/randNicknameDetails.ts
import type { NicknameDetail, RandNicknameOptions } from '../_types/global.js';
import { generateNicknameDetails } from './nicknameGenerator.js';

/**
 * Generate nicknames along with the pieces each one was built from. Takes the same
 * options as `randNickname`. Useful when you need the words on their own — to
 * highlight the base word, or to group by theme.
 *
 * @example
 * randNicknameDetails({ language: 'ko' });
 * // [{
 * //   nickname: '멋진사자',
 * //   words: ['멋진', '사자'],
 * //   language: 'ko',
 * //   theme: 'animal'
 * // }]
 */
export function randNicknameDetails(options: RandNicknameOptions = {}): NicknameDetail[] {
	return generateNicknameDetails(options);
}
